dask/config.py

```diff
@@ -114,9 +128,22 @@ def update(old: dict, new: Mapping, priority: Literal["old", "new"] = "new") ->
         if isinstance(v, Mapping):
             if k not in old or old[k] is None:
                 old[k] = {}
-            update(old[k], v, priority=priority)
+            update(
+                old[k],
+                v,
+                priority=priority,
+                defaults=defaults.get(k) if defaults else None,
+            )
         else:
-            if priority == "new" or k not in old:
+            if (
+                priority == "new"
+                or k not in old
+                or (
+                    priority == "new-defaults"
+                    and defaults
+                    and defaults.get(k) == old[k]
```

Comment:
Kind of an edge case, but I don't think this allows of us differentiate between the key not existing in `defaults` and the corresponding `old` value actually being `None` (note I've not tested the suggestion below)

 ```suggestion
                    and k in defaults
                    and defaults[k] == old[k]
```

Reply:
Thanks @jrbourbeau!